lib/config/config-cache.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * @fileoverview Responsible for caching config files
+ * @author Sylvan Mably
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Constants
+//------------------------------------------------------------------------------
+
+const VECTOR_SEP = ",";
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Get a string hash for a single object from a config vector
+ * @param {Object} entry config vector entry
+ * @returns {string} hash of the vector entry
+ * @private
+ */
+function hashVectorEntry(entry) {
+    return [].concat(entry.filePath, entry.matchingOverrides).join(VECTOR_SEP);
```

Comment:
Good catch. I've changed it to use the JSON-serialized representation of the vector as the hash.